File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/post/domain/PostApplication.java
package io.twogether.nbe_5_7_2_02team.post.domain;

import io.twogether.nbe_5_7_2_02team.member.domain.Member;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class PostApplication {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    private Post post;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "field_id", nullable = false)
    private RecruitmentField field;

    public PostApplication(Member member, Post post, RecruitmentField field) {
        this.member = member;
        this.post = post;
        this.field = field;
    }
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/post/domain/PostApplication.kt
package io.twogether.nbe_5_7_2_02team.post.domain

import io.twogether.nbe_5_7_2_02team.member.domain.Member
import jakarta.persistence.*

@Entity
class PostApplication(
    @field:JoinColumn(
        name = "member_id",
        nullable = false,
    )
    @field:ManyToOne(fetch = FetchType.LAZY)
    val member: Member,
    @field:JoinColumn(
        name = "post_id",
        nullable = false,
    )
    @field:ManyToOne(fetch = FetchType.LAZY)
    val post: Post,
    @field:JoinColumn(
        name = "field_id",
        nullable = false,
    )
    @field:ManyToOne(fetch = FetchType.LAZY)
    val field: RecruitmentField,
) {
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
}
